### DISCUSSION TITLE
issue with dynamic refresh

### DISCUSSION THREAD

**Discussion by alurisreedhar1** [Azure/CCOInsights#253] [Q&A] [answered]
Hi Team,

Please let me know if the Dynamic refresh worked for anyone so far. That would really help. I tried various changes like in the examples below but still it does not work and keeps complaining about the type of datasource in Power BI service.

1) let
 GetPages = (Path)=>
 let
     Source = Json.Document(Web.Contents(Path,[RelativePath = "/subscriptions/"&SubscriptionId & "/providers/Microsoft.Network/networkInterfaces",
												Query=[#"api-version"="2021-03-01"]] 
										)
							),
     LL= @Source[value],
     result = try @LL & @GetPages(Source[#"nextLink"]) otherwise @LL
 in
 result,
    Fullset = GetPages(
						GetManagementURL(AzureKind)

2)= let ListVirtualMachinesExtensions = (vmId as text) =>
let
    Source = Json.Document(Web.Contents(
										GetManagementURL(AzureKind),
										[RelativePath = vmId &"/extensions",
										Query=[#"api-version"="2021-07-01"]]
										)),
    value = Source[value]
   
in
    #"value"
in
    ListVirtualMachines
	
	-----------------
	= let ListVirtualMachinesExtensions = (vmId as text) =>

3) let
    url = GetManagementURL(AzureKind),
    iterations = 10,
    FuncGetOnePage = (url) as record =>
    let 
       Source = Json.Document(Web.Contents(url,[RelativePath = "/subscriptions/"&subscriptionId&"/providers/microsoft.Security/alerts",
																												Query=[#"api-version"="2021-01-01"]])),
       data = try Source[value] otherwise null,
       next = try Source[nextLink] otherwise null,
       res = [Data=data, Next=next]
    in
        res,
        GeneratedListOfPages = List.Generate(()=>[i=0, res = FuncGetOnePage(url)],
        each [i]<iterations and [res][Data]<>null,
        each [i=[i]+1, res = FuncGetOnePage([res][Next])],
        each [res][Data]),
4)

**Comment by josunefon** [comment, marked as answer]
Dynamic data soruces are not supported right now :(

We're working in a new solution (with a backend like the ADO and Github Dashboard) to provide this feature
